boards/arm/rpi_pico/doc/index.rst

```diff
@@ -133,3 +136,14 @@ Inside gdb, run:
    (gdb) file path/to/zephyr.elf
 
 You can then start debugging the board.
+
+Programmable I/O (PIO)
+**********************
+The RP2040 SoC comes with two PIO periherals. These are two simple
+co-processors that are designed for I/O operations. The PIOs run
+a custom instruction set, generated from a custom assembly language.
+PIO programs are assembled using `pioasm`, a tool provided by Raspberry Pi.
+
+Zephyr does not (currently) assemble PIO programs. Rather, they should be
+manually assembled and embedded in source code. An example of how this is done
+can be found at `drivers/serial/uart_rpi_pico_pio.c`.
```

Comment:
Its better to add a explanation about PIO UART driver. UART could be assigned any pin. (as long as 2 pins in a row)